web/modules/custom/solution/src/Plugin/Validation/Constraint/UniqueSolutionTitleValidator.php

```diff
@@ -9,6 +9,21 @@
 
 /**
  * Validates that a field is unique for the given entity type within a bundle.
+ *
+ * This is the validator for the UniqueSolutionInTitle constraint.
+ * The solutions and the releases are actually the same entity. A solution
+ * can have many releases and a release belongs to one solution. A release
+ * cannot have releases or multiple solutions.
+ *
+ * The solution entity is defined by having an empty field is_version_of.
+ * This is enough because releases can only be created through the solution and
+ * automatically have the is_version_of field filled. An entity that has the
+ * field is_version_of filled is automatically a release.
+ *
+ * The following checks make sure that a solution must have a unique title among
+ * solutions and a release must have a unique title against other solutions and
+ * their releases but can have the same name as their parent solution or their
+ * sibling releases.
```

Comment:
This is really really good documentation. It clears up all the confusion, not only for me, but for all developers dealing with this in the next 8+ years. Thanks very much!!